main.nf

```diff
@@ -591,24 +637,34 @@ workflow {
     )
 
     FEATURE_ENGINEERING_PART1 ( // will rename it once we have analyzed/review the part
-        VEP_ANNOTATE.out,
+        VEP_ANNOTATE.out.vep_output,
         HPO_SIM.out.hgmd_sim,
         HPO_SIM.out.omim_sim,
         file(params.ref_annot_dir)
     )
 
     FEATURE_ENGINEERING_PART2 (
-        FEATURE_ENGINEERING_PART1.out,
+        FEATURE_ENGINEERING_PART1.out.scores,
         PHRANK_SCORING.out,
+
         file(params.ref_annot_dir),
         file(params.ref_var_tier_dir),
-        file(params.ref_merge_expand_dir),
-        file(params.ref_mod5_diffusion_dir)
+        file(params.ref_merge_expand_dir)
+    )
+
+    MERGE_RESULTS(
+        PHRANK_SCORING.out,
+        FEATURE_ENGINEERING_PART2.out.tier.collect(),
+        FEATURE_ENGINEERING_PART2.out.compressed_scores.collect(),
+        file(params.ref_annot_dir),
+        file(params.ref_mod5_diffusion_dir),
+        file(params.ref_merge_expand_dir)
     )
 
-    PREDICTION(
-        FEATURE_ENGINEERING_PART2.out.matrix,
-        FEATURE_ENGINEERING_PART2.out.scores,
+    // Run Prediction on the final merged output
+    PREDICTION( 
+        MERGE_RESULTS.out.merged_matrix,
```

Comment:
Would you apply new name for prediction too?
```groovy
process PREDICTION {
    publishDir "${params.outdir}/prediction/", mode: "copy"
    input:
    path matrix
    path scores
    path ref_predict_new_dir
    path ref_model_inputs_dir
    output:
    path "conf_4Model/*.csv"
    path "conf_4Model/integrated/*.csv"
```

Reply:
Done.